earthly/docs/Earthfile

```diff
@@ -0,0 +1,175 @@
+VERSION 0.7
+FROM python:3.11-bullseye
+
+mkdocs-material:
+    # Derived from official mkdocs-material docker container.
+    # https://github.com/squidfunk/mkdocs-material/blob/master/Dockerfile
+
+    # What version is this
+    ARG tag=latest
+    ARG registry
+
+    # Installation tmp directory
+    WORKDIR /poetry
+
+    # Environment Variables
+    ENV POETRY_HOME=/poetry
+    ENV PATH=$POETRY_HOME/bin:$PATH
+    ENV PYTHONDONTWRITEBYTECODE=1
```

Comment:
I see this getting copy/pasted everywhere, and I understand the rationale, but it's also not _always_ required (in the case of this Earthfile, it's a no-op). I'm not opposed to keeping it, but it might be nice to get these into a separate UDC that can be pulled into Python builds. This also allows us to add more documentation around what this actually does.

Reply:
It's not a no-op.  
No problem adding it as a UDC.

The reason I had to do this is because poetry installs everything inside a `venv`,  you can't just run the installed code.  You have to `poetry run` it, or otherwise, get into the `venv` that poetry makes.  And I couldn't find a nice way to do that reliably other than `poetry run`.
But `poetry run` won't work unless you are in the same directory as the `pyproject.toml` that you installed.
So what this does is put poetry into a dependable location `/poetry`, and that's also where the `pyproject.toml` ends up.

I then had to make a shell script to actually run it, so I could still find the `mkdocs.yml` file in the mounted filesystem when the image is used live.

Actually for this image, `poetry` is a PITA.  But it's working.

Reply:
Sorry, I'm not sure why it highlighted three lines, but I was referencing only line 18:

```
ENV PYTHONDONTWRITEBYTECODE=1
```

AFAIK, since we're just using packages, this _is_ a no-op. Unless there's `.pyc` files getting generated somewhere.

Reply:
`PYTHONDONTWRITEBYTECODE=1` stops the python interpreter from saving pointless bytecode cache files inside the short-running container, which will only be lost when the container stops anyway.
It's a small perf improvement, python has the bytecode in memory anyway.

An improvement on this would be to pre-interpret all the python into bytecode and actually save it inside the container.  But that seems unnecessary.

Reply:
Yeah, I get why it makes sense for a long-running container (i.e. FastAPI), but for containers that essentially get thrown away (i.e. build and output the docs), I was thinking most of the benefits were negated. 

But yes, this is a good candidate for going into a UDT. Not sure what it would look like, but probably some sort of "Python setup" UDT that optimized the compiler in these sorts of ways. I'll see about tracking it in a separate issue.